File: parent-337/pinyougou-web-manager/src/main/java/cn/itcast/core/controller/BrandController.java
package cn.itcast.core.controller;

import cn.itcast.core.pojo.good.Brand;
import cn.itcast.core.service.BrandService;
import com.alibaba.dubbo.config.annotation.Reference;
import entity.PageResult;
import entity.Result;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;
import java.util.Map;

// 由于要响应json格式的数据,又要注册组件到容器,所以使用注解@RestController
@RestController
@RequestMapping("/brand")
public class BrandController {

    // 注入BrandService 这里要使用dubbo的注解@Reference
    @Reference
    private BrandService brandService;

    /**
     * 查询所有品牌,并显示在浏览器
     */
    @RequestMapping("/findAll")
    public List<Brand> findAll() {
        return brandService.findAll();
    }

    /**
     * 分页展示
     */
    @RequestMapping("/findPage")
    public PageResult findPage(Integer pageNum, Integer pageSize) {
        return brandService.findPage(pageNum, pageSize);
    }

    /**
     * 新增品牌
     */
    @RequestMapping("/add")
    public Result add(@RequestBody Brand brand) {
        try {
            brandService.add(brand);
            return new Result(true, "保存成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "保存失败");
        }
    }

    /**
     * 修改品牌之数据回显
     */
    @RequestMapping("/findOne")
    public Brand findOne(Long id) {
        Brand brand = brandService.findOne(id);
        return brand;
    }

    /**
     * 修改品牌之修改品牌
     */
    @RequestMapping("/update")
    public Result update(@RequestBody Brand brand) {
        try {
            brandService.update(brand);
            return new Result(true, "修改成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "修改失败");
        }
    }

    /**
     * 删除品牌
     */
    @RequestMapping("/del")
    public Result del(Long[] ids) {
        try {
            brandService.del(ids);
            return new Result(true, "删除成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "删除失败");
        }
    }

    /**
     * 条件查询
     */
    @RequestMapping("/search")
    // brand不是必须的参数,当没有条件的时候,就可以没有
    public PageResult search(Integer pageNum, Integer pageSize, @RequestBody(required = false) Brand brand) {
        return brandService.search(pageNum, pageSize, brand);
    }

    /**
     * 查询id和name,存入一个map集合 并将多个这样的集合存入list集合中
     */
    @RequestMapping("/selectOptionList")
    public List<Map> selectOptionList() {
        return brandService.selectOptionList();
    }

    /**
     * 品牌Excel表导入数据库
     */
    @RequestMapping("/ajaxUpload")
    public String ajaxUploadExcel(MultipartFile file) throws Exception {
        byte[] bytes = file.getBytes();
        return brandService.ajaxUploadExcel(bytes);
    }
}

File: parent-337/pinyougou-web-manager/src/main/webapp/js/controller/brandcontroller.js
// 利用模块获取控制器对象
app.controller('brandController', function ($scope, $controller, brandService) {

    $controller('baseController', {$scope: $scope});// 继承basecontroller

    // 查询所有品牌
    $scope.findAll = function () {
        brandService.findAll().success(
            function (response) {
                $scope.list = response;
            }
        )
    };


    // 分页查询品牌
    $scope.findPage = function (pageNum, pageSize) {
        brandService.findPage(pageNum, pageSize).success(
            function (response) {
                // 改变数据
                $scope.list = response.rows;
                // 改变总条数
                $scope.paginationConf.totalItems = response.total;
            }
        )
    };


    // 新增品牌的保存品牌
    $scope.save = function () {

        var object = null;
        if ($scope.entity.id != null) {
            object = brandService.update($scope.entity);
        } else {
            object = brandService.add($scope.entity);
        }

        object.success(
            function (response) {
                if (response.flag) {
                    alert(response.message);
                    $scope.reloadList();
                } else {
                    alert(response.message);
                }
            }
        )
    };

    // 根据id查询品牌
    $scope.findOne = function (id) {
        brandService.findOne(id).success(
            function (response) {
                $scope.entity = response;
            }
        )
    };


    // 删除指定品牌
    $scope.del = function () {
        if (confirm('你确认删除吗?')) {
           /* brandService.del($scope.selectIds).success(
                function (response) {
                    if (response.flag) {
                        alert(response.message);
                        $scope.reloadList();
                    } else {
                        alert(response.message);
                    }
                }
            )*/
        }
    };

    // 条件查询
    $scope.search = function (pageNum, pageSize) {
        brandService.search(pageNum, pageSize, $scope.searchEntity).success(
            function (response) {
                alert(1);
                console.log(response);
                // 改变数据
                $scope.list = response.rows;
                // 改变总条数
                $scope.paginationConf.totalItems = response.total;
            }
        )
    }
});